### DISCUSSION TITLE
Nginx ingress manifest persistent changes

### DISCUSSION THREAD

**Discussion by thowd22** [rancher/rke2#4126] [General]
Quick question about the nginx ingress manifest that comes bundled with RKE2.

Im using metallb with our RKE2 install and it requires some changes to /var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx.yaml . I have to add the following to the end of that yaml file to get the bundled nginx ingress to play nice with metallb. However, if one of the Control Plane, Etcd servers is rebooted for maintenance, these changes are removed and need to be reapplied. Is there a way to make these changes permanent?   

```---
apiVersion: helm.cattle.io/v1
kind: HelmChartConfig
metadata:
  name: rke2-ingress-nginx
  namespace: kube-system
spec:
  valuesContent: |-
    controller:
      publishService:
        enabled: true
      service:
        enabled: true

**Comment by Martin-Weiss** [comment]
You can create a /var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml with the above content instead of modifying the pre-delivered one... this should not be overwritten during restarts..

**Reply by thowd22** [reply]
Currently im just modifying /var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml on one of the master servers. Do I need to recreate it on every master?

**Reply by Martin-Weiss** [reply]
It is sufficient to have it on one master, only. It will create the helmchartconfig object in kube-system namespace and the helm operator will use it.
I use fleet and deploy such stuff via git ;-)
